experimental/segmenter/src/dictionary.rs

```diff
@@ -53,6 +61,17 @@ impl<'l, 's, Y: DictionaryType<'l, 's> + ?Sized> Iterator for DictionaryBreakIte
                     return Some(next.0 + Y::char_len(next.1));
                 }
                 TrieResult::Intermediate(_) => {
+                    {
+                        // If next character doesn't allow for grapheme, we don't recognize this word as intermediate state.
+                        let tmp = self.iter.clone();
+                        if let Some(tmp_next) = self.iter.next() {
```

Comment:
Suggestion: Can you use `.peek()` somehow?

Reply:
I will update to use `Peekable` instead.